internal/bus/fake_message_pipe.go

```diff
@@ -0,0 +1,135 @@
+/**
+ * Copyright (c) F5, Inc.
+ *
+ * This source code is licensed under the Apache License, Version 2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package bus
+
+import (
+	"context"
+	"testing"
+)
+
+// FakeMessagePipe is a mock message pipe
+type FakeMessagePipe struct {
+	plugins           []Plugin
+	messages          []*Message
+	processedMessages []*Message
+	ctx               context.Context
+}
+
+var _ MessagePipeInterface = &FakeMessagePipe{}
+
+func SetupFakeMessagePipe(t *testing.T, ctx context.Context, plugins []Plugin) *FakeMessagePipe {
+	messagePipe := NewFakeMessagePipe(ctx)
+
+	err := messagePipe.Register(10, plugins)
+	if err != nil {
+		t.Fail()
+	}
+	return messagePipe
+}
+
+func ValidateMessages(t *testing.T, messagePipe *FakeMessagePipe, msgTopics []string) {
+	processedMessages := messagePipe.GetProcessedMessages()
+	if len(processedMessages) != len(msgTopics) {
+		t.Fatalf("expected %d messages, received %d: %+v", len(msgTopics), len(processedMessages), processedMessages)
+	}
```

Comment:
Do asserts not work here? Or does the test not fail cause it's in a function?
```suggestion
	assert.Equal(t, len(msgTopics), len(processedMessages))
```

Reply:
Actually this function isn't used so removing it.